app/components/recirculation-prefooter.js

```diff
@@ -50,6 +51,17 @@ export default Component.extend(
       }) : this.sponsoredItem.thumbnail;
     }),
 
+    sponsoredItemObserver: observer('sponsoredItem', 'sponsoredItemDisplayed', function () {
+      if (this.sponsoredItem && this.sponsoredItemDisplayed) {
+        // this tracking needs to be done in observer since the sponsored item is fetched async
```

Comment:
hm, maybe we should return the promise from sponsored-content service to allow clients to register a handler for it

Reply:
I thought about that, but then it would be tracked when the item is fetched, not displayed. Given the fact that it is fetched only once for both, footer and search modal, then it would be tracked twice on the impression of first of the two components